lib/internal/url.js

```diff
@@ -932,6 +935,60 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
+    return `//${hostname}${pathname}`;
+  } else {
+    // Otherwise, it's a local path that requires a drive letter
+    var letter = pathname.codePointAt(1) | 0x20;
+    var sep = pathname[2];
+    if (letter < 97 || letter > 122 ||   // a..z A..Z
+        (sep !== ':' && sep !== '|')) {
+      return new Error('File URLs must specify absolute paths');
+    }
+    pathname = pathname.slice(1);
```

Comment:
Checking the first three characters is to verify that the path begins with a proper windows drive letter. Once we do that, we simply strip the leading `/` off. For instance, in the file URL `file:///c:/foo`, the `pathname` component will be `/c:/foo`, so once we verify that `c:` is a proper drive letter we use slice to drop the leading `/` making it `c:/foo`.